File: Part 2 - Regression/Section 5 - Multiple Linear Regression/multiple_regression.R
# Data Preprocessing Template

# Importing the dataset
dataset = read.csv('50_Startups.csv')


dataset$State = factor(dataset$State,
                       levels = c('California', 'Florida', 'New York'),
                       labels = c(1, 2, 3))


# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
split = sample.split(dataset$Profit, SplitRatio = 0.7)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)


regressor = lm(formula = Profit ~ R.D.Spend  ,
               data=dataset)
pred = predict(regressor,
               newdata = test_set)
summary(regressor)

File: Part 2 - Regression/Section 5 - Multiple Linear Regression/multiple_regression.py
# Data Preprocessing Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('50_Startups.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, -1].values
# Encoding categorical data

# Encoding the Independent Variable
from sklearn.preprocessing import LabelEncoder, OneHotEncoder
labelencoder_X = LabelEncoder()
X[:, 3] = labelencoder_X.fit_transform(X[:, 3])
onehotencoder = OneHotEncoder(categorical_features = [3])
X = onehotencoder.fit_transform(X).toarray()

#avoiding dummy vsriable trap

X=X[:,1:]



# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/5, random_state = 0)

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""


from sklearn.linear_model import LinearRegression
regressor=LinearRegression()
regressor.fit(X_train,y_train)


#prediction
y_pred = regressor.predict(X_test)

import statsmodels.formula.api as sm
X= np.append(np.ones((50,1),dtype=int), values=X ,axis=1)
x_opt=X[:,[0,1,2,3,4,5]]
regressor_ols = sm.OLS(endog=y, exog=x_opt).fit()
regressor_ols.summary









